Adyen/Utilities/PublicKeyProvider/PublicKeyConsumer.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+public protocol PublicKeyConsumer: PaymentComponent {
+
+    /// Provider for fetching the public key.
+    var publicKeyProvider: AnyPublicKeyProvider { get set }
+    
+    /// Convenient way to fetch the client public key with a closure for the success case
+    /// and an option to notify the delegate on the failure case.
+    /// - Parameters:
+    ///   - notifyingDelegateOnFailure: If `true`, notifies the `PaymentComponentDelegate` on failure.
+    ///   - successHandler: The block that is called when fetching was successful. Contains the public key.
+    func fetchCardPublicKey(notifyingDelegateOnFailure: Bool, successHandler: PublicKeySuccessHandler?)
```

Comment:
no need to expose this function in the protocol interface, its supposed to be a helper function for protocol adopters, its not supposed to be implemented by protocol adopters.